src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -216,10 +219,23 @@ public File newFolder() throws IOException {
     }
 
     private File createTemporaryFolderIn(File parentFolder) throws IOException {
-        File createdFolder = File.createTempFile("junit", "", parentFolder);
-        createdFolder.delete();
-        createdFolder.mkdir();
-        return createdFolder;
+        File createdFolder = null;
+        for (int i = 0; i < TEMP_DIR_ATTEMPTS; ++i) {
+            // Use createTempFile to get a suitable folder name.
+            final String suffix = ".tmp";
```

Comment:
Style-nit: we don't mark local variables as `final` unless the compiler forces us to (like when the local is used in an anonymous class).